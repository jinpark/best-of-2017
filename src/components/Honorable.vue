<template>
  <swiper-slide>
    <div class="video-info">
      <img :src="posterUrl">

      <h1 class="title">{{anime.title}}</h1>
      <div class="info">
        <button class="button is-text hidden"><icon name="wikipedia-w" scale="1.5"></icon></button>
        <h3 class="japanese-title">{{ anime.japaneseTitle }}</h3>
        <a :href=anime.wiki target="_blank" class="wiki"><button class="button is-text"><icon name="wikipedia-w" scale="1.5"></icon></button></a>
      </div>
    </div>
  </swiper-slide>
</template>

<script>

export default {
  name: 'Honorable',
  props: ['anime'],
  computed: {
    posterUrl () {
      return `/static/honorable/${this.anime.filename}.jpg`
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
h1, h2, h3 {
  font-weight: normal;
}

.video-info img {
  max-height: 50vh;
}

button.is-text {
  color: white;
  float: right;
}

.info {
  justify-content: center;
  align-items: center;
}

button.hidden {
  visibility: hidden;
  margin-left: auto;
}

a.wiki {
  margin-left: auto;
}

</style>
