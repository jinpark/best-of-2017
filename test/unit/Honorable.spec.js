import Honorable from '@/components/Honorable';
import { shallow } from 'vue-test-utils'
import { createRenderer } from 'vue-server-renderer'


describe('Honorable.vue', () => {
  const anime = {
        "title": "Welcome to the Ballroom",
        "japaneseTitle": "ボールルームへようこそ",
        "genre": "drama",
        "filename": "ballroom",
        "wiki": "https://en.wikipedia.org/wiki/Welcome_to_the_Ballroom"
  }

  it('is a swiper-slide', () => {
      const wrapper = shallow(Honorable, {propsData: {anime}})
      expect(wrapper.is('swiper-slide')).toBe(true)
  })

  it('image src is correctly computed', () => {
      const wrapper = shallow(Honorable, {propsData: {anime}})
      expect(wrapper.find('img').hasAttribute('src', '/static/honorable/ballroom.jpg')).toBe(true)
  })

  it('matches snapshot', () => {
    const renderer = createRenderer()
    const wrapper = shallow(Honorable, {propsData: {anime}})
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
