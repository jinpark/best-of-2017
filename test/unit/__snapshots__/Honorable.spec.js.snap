// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Honorable.vue matches snapshot 1`] = `
<swiper-slide>
    <div class="video-info">
        <img src="/static/honorable/ballroom.jpg">
        <h1 class="title">Welcome to the Ballroom</h1>
        <div class="info">
            <button class="button is-text hidden">
                <icon name="wikipedia-w" scale="1.5"></icon>
            </button>
            <h3 class="japanese-title">ボールルームへようこそ</h3>
            <a href="https://en.wikipedia.org/wiki/Welcome_to_the_Ballroom" target="_blank" class="wiki">
                <button class="button is-text">
                    <icon name="wikipedia-w" scale="1.5"></icon>
                </button>
            </a>
        </div>
    </div>
</swiper-slide>
`;
